x-pack/plugins/ml/public/application/components/job_spaces_repair/repair_results.tsx

```diff
@@ -0,0 +1,178 @@
+        <h3>
+          <EuiTextColor color={items.length ? 'default' : 'subdued'}>
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.savedObjectsCreated.title"
+              defaultMessage="Missing job saved objects ({count})"
+              values={{ count: items.length }}
+            />
+          </EuiTextColor>
+        </h3>
+      </EuiTitle>
+      <EuiText size="s">
+        <p>
+          <EuiTextColor color="subdued">
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.savedObjectsCreated.description"
+              defaultMessage="Jobs exist which do not have accompanying saved objects. These will be created in the current space"
+            />
+          </EuiTextColor>
+        </p>
+      </EuiText>
+    </>
+  );
+  return <RepairItem id="savedObjectsCreated" title={title} items={items} />;
+};
+
+const SavedObjectsDeleted: FC<{ repairItems: RepairSavedObjectResponse }> = ({ repairItems }) => {
+  const items = Object.keys(repairItems.savedObjectsDeleted);
+
+  const title = (
+    <>
+      <EuiTitle size="xs">
+        <h3>
+          <EuiTextColor color={items.length ? 'default' : 'subdued'}>
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.savedObjectsDeleted.title"
+              defaultMessage="Unmatched job saved objects ({count})"
+              values={{ count: items.length }}
+            />
+          </EuiTextColor>
+        </h3>
+      </EuiTitle>
+      <EuiText size="s">
+        <p>
+          <EuiTextColor color="subdued">
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.savedObjectsDeleted.description"
+              defaultMessage="Saved objects which do not have an accompanying job. These will be left over when deleting anomaly detection jobs."
+            />
+          </EuiTextColor>
+        </p>
+      </EuiText>
+    </>
+  );
+  return <RepairItem id="savedObjectsDeleted" title={title} items={items} />;
+};
+
+const DatafeedsAdded: FC<{ repairItems: RepairSavedObjectResponse }> = ({ repairItems }) => {
+  const items = Object.keys(repairItems.datafeedsAdded);
+
+  const title = (
+    <>
+      <EuiTitle size="xs">
+        <h3>
+          <EuiTextColor color={items.length ? 'default' : 'subdued'}>
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.datafeedsAdded.title"
+              defaultMessage="Job saved objects with missing datafeeds ({count})"
+              values={{ count: items.length }}
+            />
+          </EuiTextColor>
+        </h3>
+      </EuiTitle>
+      <EuiText size="s">
+        <p>
+          <EuiTextColor color="subdued">
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.datafeedsAdded.description"
+              defaultMessage="Anomaly detection job saved objects which are missing their datafeed id."
```

Comment:
```suggestion
              defaultMessage="If there are saved objects that are missing the datafeed ID for anomaly detection jobs, the ID will be added. "
```
P.S. Hopefully this doesn't assume all AD jobs must have datafeeds, since we document how to just post data to a job without requiring one.

Reply:
no, we don't assume jobs must have a datafeed. This check looks for saved objects which are missing a datafeed ID and the actual datafeed exists.
It's a situation which should never happen, but it is possible, so we need a way to repair it.

Reply:
updated in https://github.com/elastic/kibana/pull/83320/commits/29c6175bd0c1efa92d388f57dfb34a8fcb23e47e